### PR TITLE
Correctly use creatable attribute and persistability when working with domainObjects

### LINKED ISSUE #3741: Plans folder gives false sense of editability
## Overview
Deployment specific. VIPER Plans folder does not prevent editing actions such as creating and/or renaming objects. Will get persistence errors.

## Expected Behavior
Folder shouldn't be selectable for create destination. Name should not be editable.

## Steps to Reproduce
create
1. Create any object from the Create Menu
2. Select VIPER Plans Folder to save into
3. Save

rename
1. Select VIPER Plans Folder
2. Rename a plan via the browse bar

### LINKED ISSUE #4323: Fix all code that is using creatability as a proxy for persistability
#### Summary
Some of our code is using the `creatable` property of object types to determine whether an object can be persisted. Creatability and persistability are two different things. The `creatable` property only tells you whether or not an object type appears in the Create menu and nothing else. An overlay plot can be created from the Create menu, but it can also be exposed as static JSON in the tree. Using creatability to determine whether the object can be persisted in this case will give a false impression, and will result in persistence errors if any mutation or persistence is attempted.

We need to search our code base for instances where the .creatable attribute is being used to determine persistability, and replace it with a call to `openmct.object.isPersistable(domainObject)`.

Here is an example of an instance where creatability is being used instead of persistability:
https://github.com/nasa/openmct/pull/4149#discussion_r730011918

This is of fairly marginal impact right now, but it will soon become critical when we start building change-controlled display layouts and exposing them as non-editable objects in the tree.

#### Impact Check List

- [ ] Data loss or misrepresented data?
- [ ] Regression? Did this used to work or has it always been broken?
- [ ] Is there a workaround available?
- [x] Does this impact a critical component?
- [ ] Is this just a visual bug?

### LINKED ISSUE #4899: Remove NonEditableFolder plugin
**Is your feature request related to a problem? Please describe.**
Related to persistence vs creatability updates.

**Describe the solution you'd like**
Have persistence be determined by object providers.

**Describe alternatives you've considered**
This was the alternative... now it's not necessary.

**Additional context**
Directly related to:
https://github.com/akhenry/openmct-yamcs/issues/113
https://github.com/nasa/openmct/issues/4323
https://github.com/nasa/openmct/issues/3741

## Testing notes in: https://github.com/akhenry/openmct-yamcs/issues/113

### PR DESCRIPTION
**Pull request by jvigliotta**
Closes #4323 
Closes #3741
Closes #4899

### Describe your changes:
MUST be merged at the same time as: https://github.com/akhenry/openmct-yamcs/pull/112

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [ ] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [x] Unit tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate unit tests included?
* [ ] Code style and in-line documentation are appropriate?
* [ ] Commit messages meet standards?
* [ ] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

**Comment by jvigliotta** [comment]
@scottbell to take a look

File: src/plugins/linkAction/LinkAction.js [hunk 1]
```diff
@@ -33,10 +33,11 @@ export default class LinkAction {
     }
 
     appliesTo(objectPath) {
-        let domainObject = objectPath[0];
-        let type = domainObject && this.openmct.types.get(domainObject.type);
+        // let domainObject = objectPath[0];
```
Comment (scottbell) [inline src/plugins/linkAction/LinkAction.js:36]: should we remove the commented out code here?

**Review by scottbell — CHANGES_REQUESTED** [review]
A few questions, but otherwise looks good!

File: src/exporters/CSVExporter.js [hunk 2]
```diff
@@ -21,7 +21,7 @@
  *****************************************************************************/
 
 import CSV from 'comma-separated-values';
-import {saveAs} from 'saveAs';
+import { saveAs } from 'file-saver';
```
[the remark this answers is not in the document]
Comment (jvigliotta) [reply src/exporters/CSVExporter.js:24]: Yep! Seems to be fine now, I'll revert. Thanks!

File: src/plugins/linkAction/LinkAction.js (hunk 1 above)
Comment (jvigliotta) [reply src/plugins/linkAction/LinkAction.js:36]: Yep!

**Review by scottbell — APPROVED** [review]

**Pull request merged**